sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/NativeMethodsShouldBeWrapped.cs

```diff
@@ -108,24 +121,18 @@ private static void ReportTrivialWrappers(SyntaxNodeAnalysisContext c)
                 .Where(IsExternal)
                 .ToHashSet();
 
-        private static bool HasTwoOrMoreStatements(MethodDeclarationSyntax methodDeclaration) =>
-            GetBodyDescendants(methodDeclaration)
-                .OfType<StatementSyntax>()
-                .Take(2)
-                .Count() == 2;
-
         private static IEnumerable<SyntaxNode> GetBodyDescendants(MethodDeclarationSyntax methodDeclaration) =>
             methodDeclaration.Body?.DescendantNodes()
             ?? methodDeclaration.ExpressionBody?.DescendantNodes()
             ?? Enumerable.Empty<SyntaxNode>();
 
-        private static bool IsPublicOrProtected(SyntaxToken token) =>
-            token.IsKind(SyntaxKind.PublicKeyword) || token.IsKind(SyntaxKind.ProtectedKeyword);
-
-        private static bool IsExternal(SyntaxToken token) =>
-            token.IsKind(SyntaxKind.ExternKeyword);
+        private static bool IsPubliclyAccessible(IMethodSymbol methodSymbol) =>
+            PublicOrProtected.Contains(methodSymbol.GetEffectiveAccessibility());
 
         private static bool IsExternal(IMethodSymbol symbol) =>

```

Comment:
Why not just use the property instead of wrapping?

Reply:
because Linq looks a bit better with a method than a property (see GetExternalMethods)